File: app/src/main/java/com/sesame/personalhomepage/repository/UserRepository.java
package com.sesame.personalhomepage.repository;

import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import com.sesame.personalhomepage.api.Api;
import com.sesame.personalhomepage.db.UserDao;
import com.sesame.personalhomepage.model.User;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class UserRepository {
    private UserDao mUserDao;
    private Api mApi;

    public UserRepository(UserDao userDao, Api api) {
        mUserDao = userDao;
        mApi = api;
    }

    public LiveData<User> getUser(String name) {
        refresh(name);
        return mUserDao.getUserByLoginName(name);
    }

    public void refresh(String userName) {
        mApi.getUser(userName).enqueue(new Callback<User>() {
            @Override
            public void onResponse(Call<User> call, Response<User> response) {
                User user = response.body();
                if (user != null) {
                    insertUser(user);
                }
            }

            @Override
            public void onFailure(Call<User> call, Throwable t) {
                t.printStackTrace();
            }
        });

    }

    private void insertUser(User user) {
        AsyncTask.execute(new Runnable() {
            @Override
            public void run() {
                if (user != null) {
                    mUserDao.insertUser(user);
                }
            }
        });
    }
}

File: app/src/main/java/com/sesame/personalhomepage/repository/UserRepository.kt
package com.sesame.personalhomepage.repository

import android.app.Application
import android.widget.Toast
import androidx.lifecycle.MutableLiveData
import com.sesame.personalhomepage.model.User
import com.sesame.personalhomepage.network.ApiService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class UserRepository(
    val application: Application,
    private val apiService: ApiService
) {

    fun requestFromApi(
        user: String,
        liveData: MutableLiveData<User>
    ) {
        apiService.getUser(user).enqueue(object : Callback<User> {
            override fun onResponse(call: Call<User>, response: Response<User>) {
                liveData.value = response.body()
            }

            override fun onFailure(call: Call<User>, t: Throwable) {
                Toast.makeText(application, "onFailure", Toast.LENGTH_SHORT).show()
            }
        })
    }

}